src/cobalt.rs

```diff
@@ -181,40 +201,43 @@ fn generate_pages(posts: Vec<Document>, documents: Vec<Document>, context: &Cont
     trace!("Generating other documents");
     for mut doc in documents {
         trace!("Generating {}", doc.url_path);
-        if doc.front.pagination.is_some() {
-            let paginators = pagination::generate_paginators(&mut doc, &posts_data)?;
-            // page 1 uses frontmatter.permalink instead of paginator.permalink
-            let mut paginators = paginators.into_iter();
-            let paginator = paginators
-                .next()
-                .expect("We detected pagination enabled but we have no paginator");
+        if doc.front.pagination.order == SortOrder::None {
```

Comment:
Note to self: double check the change away from an `Option`

Reply:
@Geobert could you explain your theory of ops here?  It seems like the paginator is always enabled for all pages, regardless of what collection they are in (posts or pages) when
- I thought pagination is only supported (and so should error otherwise) in pages
- The goal was to be compatible while allowing people to opt-out.  It doesn't look like this provides an opt-out mechanism.

My thought for how this would have been implemented is to modify the `pages` specific defaults to set an active paginator over `all`.  That makes it easy for people to opt-out on a global level while then opting in on a per-page level.

Reply:
1. You're right, I messed up, going back to that code.
2. Opt-out is done like this in `_cobalt.yml`:
```yml
pages:
  default:
    pagination_compat: false
```

Reply:
Trying to sort 1., I'm wondering if the `pagination_compat` flag in the frontmatter is a good idea. Maybe only a global flag in `PageConfig` would make more sense?

Reply:
I'm mixed.
- Global flag would make sense for the `collections`, previous, and next support in posts
- Somehow we still need to pass the result of that global flag to the processing of each page
- Would it be useful for someone to disable it for all pages and then opt-back-in on a page-by-page basis until they migrate?

Reply:
> Somehow we still need to pass the result of that global flag to the processing of each page

The flag would be in `context.pages`, wouldn't it?

> Would it be useful for someone to disable it for all pages and then opt-back-in on a page-by-page basis until they migrate?

As the legacy support is on by default, I can't see why someone would deactivate it before migrating all to `paginator`.

Reply:
Sorry for the late update, holidays are finished so less time (and I'm trying to get involved in rust-analyzer as well) Shall I proceed with a global flag?